### PR TITLE
Unix.accept may raise an exception

### PR DESCRIPTION
**Pull request by craff**
solves issue #21

### CODE REVIEW

File: src/Tiny_httpd.ml [hunk 1]
```diff
@@ -1088,17 +1088,21 @@ let run (self:t) : (unit,_) result =
...
+              handle_client_ self client_sock;
+              Sem_.release 1 self.sem_max_connections;
+            with e ->
+              (try Unix.close client_sock with _ -> ());
+              Sem_.release 1 self.sem_max_connections;
+              raise e
+          );
+      with e ->
+        Printf.eprintf "accept raised an exception: %s"
+           (Printexc.to_string e)
```
Comment (c-cube) [inline src/Tiny_httpd.ml:1105]:
```suggestion
        _debug (fun k->k "accept raised an exception: %s" (Printexc.to_string e))
```

Comment (c-cube) [inline src/Tiny_httpd.ml]:
```suggestion
      match Unix.accept sock with
      | client_sock, _ ->
        self.new_thread
          (fun () ->
            try
              handle_client_ self client_sock;
              Sem_.release 1 self.sem_max_connections;
            with e ->
              (try Unix.close client_sock with _ -> ());
              Sem_.release 1 self.sem_max_connections;
              raise e
          );
      | exception e ->
        Printf.eprintf "accept raised an exception: %s"
           (Printexc.to_string e)
```

Comment (c-cube) [reply src/Tiny_httpd.ml]: this way only exceptions raised in `accept` will be captured.

Comment (craff) [reply src/Tiny_httpd.ml]: Thread.create may raise exception too (ok, the documentation does not mention it, but there is probably a maximum number of threads?) and exception in the son thread will never reach the father thread.
If you want to keep the server alive in a burst of request reaching the maximum number of thread temporarely, 
the best is to catch both exception from Unix.accept and Thread.create ?

**Comment by craff** [comment]
Sorry, I did not see your commit ... so a useless merge is in the commit ...
I prefer the current version, it lowers the risk of server crash.

**Comment by craff** [comment]
A very pedantic version would tell in the message if the exception comes from accept or Thread.create. Usefull?

**Comment by c-cube** [comment]
Ok, let's take the safe way :)

**Pull request merged**